src/V3AstNodes.cpp

```diff
@@ -2326,6 +2387,29 @@ void AstCAwait::dump(std::ostream& str) const {
         sensesp()->dump(str);
     }
 }
+bool AstCMethodHard::isPure() {
+    if (!m_purity.isCached()) m_purity.setPurity(getPurity());
+    return m_purity.isPure();
+}
+const char* AstCMethodHard::broken() const {
+    BROKEN_RTN(m_purity.isCached() && m_purity.isPure() != getPurity());
+    return nullptr;
+}
+bool AstCMethodHard::getPurity() const {
+    static const char* pureMethods[] = {"size",        "exists",
+                                        "at",          "atBack",
+                                        "find",        "find_index",
+                                        "find_frst",   "find_first_index",
+                                        "find_last",   "find_last_index",
+                                        "min",         "max",
+                                        "unique",      "unique_index",
+                                        "sum",         "product",
+                                        "and",         "or",
+                                        "xor",         "isFired",
+                                        "isTriggered", "any"};
```

Comment:
sort the list.
If we're going to do it this way, which I'm ok with, I think we should list absolutely every method and assert if not found, as otherwise we'll add a new one and forget to set the purity correctly.

Reply:
Thanks. I added it to broken() check. I skipped `void` methods, because they are always impure (they should be removed otherwise).